internal/alias/target/repository_alias_list_resolvable.go

```diff
@@ -41,7 +45,50 @@ func (r *Repository) listResolvableAliases(ctx context.Context, permissions []pe
 	case len(permissions) == 0:
 		return nil, time.Time{}, errors.New(ctx, errors.InvalidParameter, op, "missing permissions")
 	}
-	toTargetIds, toTargetsInScopeIds := targetAndScopeIdsForDestinations(permissions)
+
+	var allDescendants bool
+	// First check for all descendants. Since what we are querying for below is
+	// for targets (either IDs, or targets within specific scopes), and targets
+	// are not in global, if this matches we can actually ignore everything
+	// else.
+	for _, perm := range permissions {
+		if perm.GrantScopeId == globals.GrantScopeDescendants && perm.All {
+			allDescendants = true
+		}
+	}
+
+	directIds := make([]string, 0, len(permissions))
+	directScopeIds := make([]string, 0, len(permissions))
+	childAllScopes := make([]string, 0, len(permissions))
+	for _, perm := range permissions {
+		switch {
+		case allDescendants:
+			// See the above check; we don't need any other info
+		case perm.GrantScopeId == scope.Global.String() || strings.HasPrefix(perm.GrantScopeId, globals.OrgPrefix):
+			// There are no targets in global or orgs
+		case perm.RoleScopeId == scope.Global.String() && perm.GrantScopeId == globals.GrantScopeChildren:
+			// A role in global that includes children will include only orgs,
+			// which do not have targets, so ignore
+		case perm.GrantScopeId == globals.GrantScopeChildren && perm.All:
+			// Because of the above check this will match only grants from org
+			// roles. If the grant scope is children and all, we store the scope
+			// ID.
+			childAllScopes = append(childAllScopes, perm.RoleScopeId)
+		case perm.All:
+			// We ignore descendants and if this was a children grant scope and
+			// perm.All it would match the above case. So this is a grant
+			// directly on a scope. Since only projects contain targets, we can
+			// ignore any grant scope ID that doesn't match targets.
+			if strings.HasPrefix(perm.GrantScopeId, globals.ProjectPrefix) {
+				directScopeIds = append(directScopeIds, perm.GrantScopeId)
+			}
+		case len(perm.ResourceIds) > 0:
+			// It's an ID grant
+			directIds = append(directIds, perm.ResourceIds...)
+		}
+	}
+
+	log.Println("allDescendants", allDescendants, "childAllScopes", pretty.Sprint(childAllScopes), "direct ids", pretty.Sprint(directIds), "direct scope ids", pretty.Sprint(directScopeIds))
```

Comment:
I plan to -- will leave this comment here as a reminder -- but since the resolvable alias tests aren't done I want to keep it in for now to help with debugging there.

Reply:
Done.